website/blog/2025-04-20-vs-code-with-podman-desktop.md

```diff
@@ -0,0 +1,100 @@
+
+Let's install them one by one.
+
+### Option 1: Microsoft’s Container Tools extension
+
+To install the extension:
+
+1. Open VS Code.
+2. Click on the Extensions icon in the Activity Bar (or press `Ctrl+Shift+X` or `Cmd+Shift+X`).
+3. Search for "Container Tools" and install the extension by Microsoft.
+
+#### Configuring VS Code for Podman
+
+The Container Tools extension usually automatically detects Podman if Docker isn't running, by looking at the `DOCKER_HOST` environment variable. In Podman Desktop navigate to Settings \> Docker Compatibility \> Third-Party Tool Compatibility and make sure the option is enabled. It’s possible that if you installed through `brew` that this is not correctly set.
+
+![enabling docker compatibility in the settings](img/vs-code-podman/docker-compatibility.png)
+
+### Option 2: Pod Manager
+
+To install the extension:
+
+1. Open VS Code.
+2. Click on the Extensions icon in the Activity Bar (or press `Ctrl+Shift+X` or `Cmd+Shift+X`).
+3. Search for "Pod Manager" and install the extension by dreamcatcher45.
+
+## Using the VS Code Extensions
+
+Now that we have the extensions installed and configured, let's see how to use them.
+
+### Working with Containerfiles and Dockerfiles
+
+Both extensions provide syntax highlighting, code completion, and linting for Containerfiles and Dockerfiles. Open a Containerfile in VS Code, and you'll immediately benefit from these features.
+
+You can also build images directly from VS Code:
+
+1. Right-click on the Containerfile in the Explorer view.
+2. Select "Build Image".
+3. VS Code will prompt you for an image name and tag.
+4. The extension will then build the image using Podman.
+
+![building a Containerfile in vs code using the microsoft extension](img/vs-code-podman/build-image.png)
+
+### Managing Containers
+
+The extensions also allow you to manage containers directly from VS Code. You can start, stop, restart, and remove containers, as well as view their logs and inspect their configuration.
+
+To view the container logs:
+
+1. Click on the Container icon in the Activity Bar.
+2. You'll see a list of your containers, images, and networks.
+3. Right-click on a container to perform actions.
+
+![using Container Tools extension to view the logs of the container](img/vs-code-podman/view-logs.png)
+
+Similarly using the Pod manager extension we can visually inspect containers, images, and volumes.
+
+1. Click on the Pod manager icon in the Activity Bar.
+2. You'll see a list of your containers, images, and networks.
+
+![using podmanager to view all the running containers, images and volumes](img/vs-code-podman/podmanager-details.png)
+
+and manage the container lifecycle.
+
+![using podmanager to manage the lifecycle of the container](img/vs-code-podman/interact-with-container.png)
+
+and of course troubleshoot issues with a visual interface.
+
+![using podmanager to enter the container](img/vs-code-podman/podmanager-details.png)
+
+## Conclusion
```

Comment:
Where to report issues? Inviting people to share their experiences?

Reply:
Added